diff --git a/node_modules/exceljs/lib/xlsx/xform/book/workbook-xform.js b/node_modules/exceljs/lib/xlsx/xform/book/workbook-xform.js
index cab6bf4..8f4ec67 100755
--- a/node_modules/exceljs/lib/xlsx/xform/book/workbook-xform.js
+++ b/node_modules/exceljs/lib/xlsx/xform/book/workbook-xform.js
@@ -42,15 +42,18 @@ class WorkbookXform extends BaseXform {
     let index = 0; // sheets is sparse array - calc index manually
     model.sheets.forEach(sheet => {
       if (sheet.pageSetup && sheet.pageSetup.printArea) {
+        const definedName = {
+          name: '_xlnm.Print_Area',
+          ranges: [],
+          localSheetId: index,
+        };
         sheet.pageSetup.printArea.split('&&').forEach(printArea => {
           const printAreaComponents = printArea.split(':');
-          const definedName = {
-            name: '_xlnm.Print_Area',
-            ranges: [`'${sheet.name}'!$${printAreaComponents[0]}:$${printAreaComponents[1]}`],
-            localSheetId: index,
-          };
-          printAreas.push(definedName);
+          definedName.ranges.push(
+            `'${sheet.name}'!$${printAreaComponents[0]}:$${printAreaComponents[1]}`
+          );
         });
+        printAreas.push(definedName);
       }
 
       if (
@@ -193,10 +196,12 @@ class WorkbookXform extends BaseXform {
           if (!worksheet.pageSetup) {
             worksheet.pageSetup = {};
           }
-          const range = colCache.decodeEx(definedName.ranges[0]);
-          worksheet.pageSetup.printArea = worksheet.pageSetup.printArea
-            ? `${worksheet.pageSetup.printArea}&&${range.dimensions}`
-            : range.dimensions;
+          for (let i = 0; i < definedName.ranges.length; ++i) {
+            const range = colCache.decodeEx(definedName.ranges[i]);
+            worksheet.pageSetup.printArea = worksheet.pageSetup.printArea
+              ? `${worksheet.pageSetup.printArea}&&${range.dimensions}`
+              : range.dimensions;
+          }
         }
       } else if (definedName.name === '_xlnm.Print_Titles') {
         worksheet = worksheets[definedName.localSheetId];
